src/PoolManager.sol

```diff
@@ -377,15 +377,23 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
     }
 
     /// @inheritdoc IPoolManager
-    function mint(Currency currency, address to, uint256 amount) external noDelegateCall onlyByLocker {
+    function mint(Currency currency, address to, uint256 amount) external override noDelegateCall onlyByLocker {
         _accountDelta(currency, amount.toInt128());
-        _mint(to, currency, amount);
+        _mint(to, currency.toId(), amount);
     }
 
     /// @inheritdoc IPoolManager
-    function burn(Currency currency, uint256 amount) external noDelegateCall onlyByLocker {
+    function burn(Currency currency, address from, uint256 amount) external override noDelegateCall onlyByLocker {
         _accountDelta(currency, -(amount.toInt128()));
-        _burn(currency, amount);
+        uint256 id = currency.toId();
+        address sender = msg.sender;
+        if (from != sender && !isOperator[from][sender]) {
```

Comment:
i think id rather this be in a helper contract - feels weird that PM is having to deal with allowances and operators when thats a token-specific thing
maybe we could have an intermediate contract like `V4Core6909` or something that is `6909` but adds `burnFrom` logic?

Reply:
Yea agree, was kinda thinking that too. Will implement